File: ProxiCall/Console_Speech/Services/Speech/TextToSpeech.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Net.Http;
using System.Text;
using System.Threading;
using System.Threading.Tasks;

namespace Console_Speech.Services.Speech
{
    class TextToSpeech
    {
        private static readonly string host = "https://westeurope.tts.speech.microsoft.com/cognitiveservices/v1";

        public static async Task<byte[]> TransformTextToSpeechAsync(string texttotransform, string locale)
        {
            // Gets an access token
            string accessToken;

            // Add your subscription key here
            // If your resource isn't in WEST US, change the endpoint
            AuthentificationSpeechApi auth = new AuthentificationSpeechApi(Environment.GetEnvironmentVariable("SpeechApiKey"));
            accessToken = auth.GetAccessToken();

            // Set request body
            string body = @"<speak version='1.0' xmlns='http://www.w3.org/2001/10/synthesis' xml:lang='" + locale + "'>" +
                                "<voice name='Microsoft Server Speech Text to Speech Voice (en-US, JessaNeural)'>" +
                                    texttotransform +
                                "</voice>" +
                            "</speak>";

            // Http request
            using (var httpClient = new HttpClient())
            {
                using (var request = new HttpRequestMessage())
                {
                    // Set the HTTP method
                    request.Method = HttpMethod.Post;
                    // Construct the URI
                    request.RequestUri = new Uri(host);
                    // Set the content type header
                    request.Content = new StringContent(body, Encoding.UTF8, "application/ssml+xml");
                    // Set additional header, such as Authorization and User-Agent
                    request.Headers.Add("Authorization", "Bearer " + accessToken);
                    request.Headers.Add("Connection", "Keep-Alive");
                    // Update your resource name
                    request.Headers.Add("User-Agent", "ProxiCallSpeech");
                    request.Headers.Add("X-Microsoft-OutputFormat", "riff-16khz-16bit-mono-pcm");
                    // Create a request
                    using (var response = await httpClient.SendAsync(request))
                    {
                        response.EnsureSuccessStatusCode();
                        var audioResponse = await response.Content.ReadAsByteArrayAsync();
                        return audioResponse;
                    }
                }
            }
        }
    }
}

File: ProxiCall.NodeJS/ProxiCall.Web/services/text-to-speech.js
'use strict';
// Requires request for HTTP requests
const request = require('request');
// Requires xmlbuilder to build the SSML body
const xmlbuilder = require('xmlbuilder');

// Requires dotenv to read .env
const dotenv = require('dotenv');
dotenv.config();

const subscriptionKey = process.env.SPEECH_KEY;
if (!subscriptionKey) {
    throw new Error('Environment variable for your subscription key is not set.')
};

const subscriptionRegion = process.env.SPEECH_Region;
if (!subscriptionRegion) {
    throw new Error('Environment variable for your subscription region is not set.')
};

var byteArray;

function textToSpeech(subscriptionKey, subscriptionRegion, saveAudio) {
    let options = {
        method: 'POST',
        uri: `https://westeurope.api.cognitive.microsoft.com/sts/v1.0/issueToken`,
        headers: {
            'Ocp-Apim-Subscription-Key': subscriptionKey
        }
    };
    // This function retrieve the access token and is passed as callback
    // to request below.
    function getToken(error, response, body) {
        console.log("Getting your token...\n")
        if (!error && response.statusCode == 200) {
            //This is the callback to our saveAudio function.
            // It takes a single argument, which is the returned accessToken.
            saveAudio(body)
        }
        else {
            throw new Error(error);
        }
    }
    request(options, getToken)
}

/* Make sure to update User-Agent with the name of your resource.
   You can also change the voice and output formats. See:
   https://docs.microsoft.com/azure/cognitive-services/speech-service/language-support#text-to-speech */
function saveAudio(accessToken) {
    // Create the SSML request.
    let xml_body = xmlbuilder.create('speak')
        .att('version', '1.0')
        .att('xml:lang', 'en-us')
        .ele('voice')
        .att('xml:lang', 'en-us')
        .att('name', 'Microsoft Server Speech Text to Speech Voice (en-US, Guy24KRUS)')
        .txt("test")
        .end();
    // Convert the XML into a string to send in the TTS request.
    let body = xml_body.toString();

    /* This sample assumes your resource was created in the WEST US region. If you
       are using a different region, please update the uri. */
    let options = {
        method: 'POST',
        baseUrl: 'https://westeurope.tts.speech.microsoft.com/',
        url: 'cognitiveservices/v1',
        headers: {
            'Authorization': 'Bearer ' + accessToken,
            'cache-control': 'no-cache',
            'User-Agent': 'YOUR_RESOURCE_NAME',
            'X-Microsoft-OutputFormat': 'riff-24khz-16bit-mono-pcm',
            'Content-Type': 'application/ssml+xml'
        },
        body: body
    };
    /* This function makes the request to convert speech to text.
       The speech is returned as the response. */
    function convertText(error, response, body) {
        if (!error && response.statusCode == 200) {
            console.log("Converting text-to-speech. Please hold...\n")
        }
        else {
            throw new Error(error);
        }
        console.log("Your file is ready.\n");
        var buf = Buffer.from(body);
        byteArray = buf.readUInt8(0);
    }
    request(options, convertText);
}

function runTextToSpeech() {
    textToSpeech(process.env.SPEECH_KEY, 'westeurope', saveAudio);
    return byteArray;
}

module.exports.runTextToSpeech = runTextToSpeech;